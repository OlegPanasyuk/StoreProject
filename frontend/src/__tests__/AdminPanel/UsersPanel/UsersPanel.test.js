import React from 'react';
import UsersPanelConnected, { UsersPanel } from '../../../AdminPanel/UsersPanel/UsersPanel';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    showUsers: () => { },
    filters: {

    },
    filterUsers: () => { },
    userToEdit: {},
    usersToShow: [],
    usersToShowCount: 0,
    editUserClose: () => { }
}) {
    let component = shallow(<UsersPanel {...props} />);
    return { component, props };
}

describe('UsersPanel --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(UsersPanel.propTypes.usersToShow).toBe(PropTypes.array);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('UsersPanel ', () => {
    let initialState = {
        adminPanel_usersPanel: {
            usersToShow: {
                rows: [],
                count: 0
            },
            userToEdit: {
                username: 'User'
            },
            filters: {

            }
        }
    };

    const mockStore = configureStore();
    let store, wrapper;
   
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><UsersPanelConnected  /></Provider>);
    });
    it('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(UsersPanelConnected).length).toEqual(1);
    });
});

