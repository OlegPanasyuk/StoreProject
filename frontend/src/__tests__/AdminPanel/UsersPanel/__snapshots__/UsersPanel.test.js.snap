// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPanel --- Snapshot Component must be rendered 1`] = `
<Container
  as="div"
  fluid={false}
>
  <Bootstrap(Row)>
    <Col
      as="div"
    >
      <ListOfPages
        count={0}
        openPage={[Function]}
      />
    </Col>
  </Bootstrap(Row)>
  <Bootstrap(Row)>
    <Col
      as="div"
      className="col-3"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add User
      </Button>
      <ConnectFunction
        updateState={[Function]}
      />
    </Col>
    <Col
      as="div"
      className="col-9"
    >
      <CardColumns />
    </Col>
  </Bootstrap(Row)>
</Container>
`;
